### PR TITLE
Use diamond operator in jdt.core

### PR DESCRIPTION
**Pull request by vogella**
Using the JDT UI clean-up, this removes the redundant type information in the jdt.core plug-in

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by vogella** [comment]
@jukzi WDYT?

**Comment by jukzi** [comment]
> Shall the activation of this a error / warning do into this commit or another PR?

into this commit

**Comment by vogella** [comment]
> > Why does JDT core not validate on all platform via its GH actions like platform and PDE? According to some test result pages which I have seen jdt tests are flunky on some platforms, e.g. https://download.eclipse.org/eclipse/downloads/drops4/I20231128-1800/testResults.php
> 
> i guess JDT depends more on java versions then on OS

I this case I would expect GH testing multiple Java versions.

**Comment by iloveeclipse** [comment]
> I guess but I can send you a PR with 130 additional files + config changes

Please do it by bundle / not with ~100 files. We had enough regressions slipped through PR's with mass changes.

**Comment by jukzi** [comment]
> Easier to review, I guess but I can send you a PR with 130 additional files + config changes

i think its better to have a single or very few PRs then for each plugin. 200 file is reviewable

**Comment by vogella** [comment]
I'm getting different messages from @jukzi and @iloveeclipse

As @jukzi seems to be interested in actually reviewing, I followed his request. In case this leads @iloveeclipse to block, I can also split up by plug-in. With Git this only cost me a minute or so, so feel free to change your requirements.

**Comment by vogella** [comment]
Give me a short while, while I review this here at GH myself, the compare editor in Eclipse is not so nice for reviewing such large changes (file by file comparison vrs all here at GH)

**Comment by jukzi** [comment]
> Give me a short while, while I review this here at GH myself, the compare editor in Eclipse is not so nice for reviewing such large changes (file by file comparison vrs all here at GH)

GH so slow... sometimes i create a diff and review it with notepad instead :-)

**Comment by vogella** [comment]
> > Give me a short while, while I review this here at GH myself, the compare editor in Eclipse is not so nice for reviewing such large changes (file by file comparison vrs all here at GH)
> 
> GH so slow... sometimes i create a diff and review it with notepad instead :-)

Sad that two Eclipse committer find the Eclipse IDE not good enough to compare (mass) changes.

**Comment by vogella** [comment]
Change looks good to me, I was mainly scanning for lost comments, the JDT clean-ups are known to have bugs wrt. comments.

Glad to see that no version change seems necessary, I guess I have to thank @HannesWell StringBuffer change for that.

**Comment by vogella** [comment]
> Give me a short while, while I review this here at GH myself, the compare editor in Eclipse is not so nice for reviewing such large changes (file by file comparison vrs all here at GH)

Ready for a review by @jukzi

File: org.eclipse.jdt.annotation/.settings/org.eclipse.jdt.core.prefs [hunk 1]
```diff
@@ -71,38 +78,47 @@ org.eclipse.jdt.core.compiler.problem.noEffectAssignment=warning
...
 org.eclipse.jdt.core.compiler.problem.parameterAssignment=ignore
+org.eclipse.jdt.core.compiler.problem.pessimisticNullAnalysisForFreeTypeVariables=warning
 org.eclipse.jdt.core.compiler.problem.possibleAccidentalBooleanAssignment=warning
 org.eclipse.jdt.core.compiler.problem.potentialNullReference=error
 org.eclipse.jdt.core.compiler.problem.potentiallyUnclosedCloseable=warning
 org.eclipse.jdt.core.compiler.problem.rawTypeReference=warning
 org.eclipse.jdt.core.compiler.problem.redundantNullAnnotation=warning
 org.eclipse.jdt.core.compiler.problem.redundantNullCheck=warning
-org.eclipse.jdt.core.compiler.problem.redundantSpecificationOfTypeArguments=warning
+org.eclipse.jdt.core.compiler.problem.redundantSpecificationOfTypeArguments=error
```
[the remark this answers is not in the document]
Comment (vogella) [reply org.eclipse.jdt.annotation/.settings/org.eclipse.jdt.core.prefs]: Done

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,6 @@ Snap.*
 # maven
 /*/target/
 /target/
+
+# pomless
```
[the remark this answers is not in the document]
Comment (vogella) [reply .gitignore]: Removed

File: org.eclipse.jdt.apt.pluggable.tests/.settings/org.eclipse.jdt.core.prefs [hunk 3]
```diff
@@ -81,21 +89,28 @@ org.eclipse.jdt.core.compiler.problem.specialParameterHidingField=disabled
...
 org.eclipse.jdt.core.compiler.problem.suppressOptionalErrors=disabled
 org.eclipse.jdt.core.compiler.problem.suppressWarnings=enabled
+org.eclipse.jdt.core.compiler.problem.suppressWarningsNotFullyAnalysed=info
 org.eclipse.jdt.core.compiler.problem.syntacticNullAnalysisForFields=disabled
 org.eclipse.jdt.core.compiler.problem.syntheticAccessEmulation=ignore
+org.eclipse.jdt.core.compiler.problem.terminalDeprecation=warning
 org.eclipse.jdt.core.compiler.problem.typeParameterHiding=warning
 org.eclipse.jdt.core.compiler.problem.unavoidableGenericTypeProblems=enabled
-org.eclipse.jdt.core.compiler.problem.uncheckedTypeOperation=warning
+org.eclipse.jdt.core.compiler.problem.uncheckedTypeOperation=info
```
Comment (jukzi) [inline org.eclipse.jdt.apt.pluggable.tests/.settings/org.eclipse.jdt.core.prefs:98]: unrelated change?

File: org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/WeakHashSet.java [hunk 4]
```diff
@@ -170,7 +170,7 @@ public T get(T obj) {
 	}
 
 	private void rehash() {
-		WeakHashSet<T> newHashSet = new WeakHashSet<T>(this.elementSize * 2);		// double the number of expected elements
+		WeakHashSet<T> newHashSet = new WeakHashSet<>(this.elementSize * 2);		// double the number of expected elements
```
Comment (jukzi) [inline org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/WeakHashSet.java:173]: intersting: my own code modernized :-)

**Review by jukzi — CHANGES_REQUESTED** [review]
A single option seems wrong, everything else straight forward. have a nice weekend

File: org.eclipse.jdt.apt.pluggable.tests/.settings/org.eclipse.jdt.core.prefs (hunk 3 above)
Comment (vogella) [reply org.eclipse.jdt.apt.pluggable.tests/.settings/org.eclipse.jdt.core.prefs]: Undone the unrelated change

**Pull request merged**